modules/simple-payments/simple-payments.php

```diff
@@ -0,0 +1,191 @@
+	static $post_type_product = 'jp_pay_product';
+
+	static $shortcode = 'simple-payment';
+
+	// Classic singleton pattern:
+	private static $instance;
+	private function __construct() {}
+	static function getInstance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+			self::$instance->register_init_hook();
+		}
+		return self::$instance;
+	}
+
+	private function register_scripts() {
+		wp_register_script( 'paypal-checkout-js', 'https://www.paypalobjects.com/api/checkout.js' );
+		wp_register_script( 'paypal-express-checkout', plugins_url( '/paypal-express-checkout.js', __FILE__ ) , array( 'paypal-checkout-js' ) );
+	}
+	private function register_init_hook() {
+		add_action( 'init', array( $this, 'init_hook_action' ) );
+	}
+	private function register_shortcode() {
+		add_shortcode( static::$shortcode, array( $this, 'parse_shortcode' ) );
+	}
+
+	public function init_hook_action() {
+		add_filter( 'rest_api_allowed_post_types', array( $this, 'allow_rest_api_types' ) );
+		$this->register_scripts();
+		$this->register_shortcode();
+		$this->setup_cpts();
+	}
+
+	function parse_shortcode( $attrs, $content = false ) {
+		if( empty( $attrs[ 'id' ] ) ) {
+			return;
+		}
+		$product = get_post( $attrs[ 'id' ] );
+		if( is_wp_error( $product ) ) {
+			return;
+		}
+		if( $product->post_type !== self::$post_type_product ) {
+			return;
+		}
+
+		// We allow for overriding the presentation labels
+		$data = shortcode_atts( array(
+			'dom_id' => uniqid( 'jp_simple_payments__button_' . $product->ID . '_' ),
+			'class' => 'jp_simple_payments__' . $product->ID,
+			'title' => get_the_title( $product ),
+			'description' => $product->post_content,
+			'cta' => get_post_meta( $product->ID, 'spay_cta', true ),
+		), $attrs );
+		$data[ 'price' ] = $this->format_price(
+			get_post_meta( $product->ID, 'spay_price', true ),
+			get_post_meta( $product->ID, 'spay_currency', true ),
+			$data
+		);
+
+		wp_enqueue_script( 'paypal-express-checkout' );
+		wp_add_inline_script( 'paypal-express-checkout', "try{PaypalExpressCheckout.renderButton( {$data['dom_id']} );}catch(e){}" );
+
+		return $this->output_shortcode( $data );
+	}
+
+	function output_shortcode( $data ) {
+		$output = <<<TEMPLATE
+<div class="{$data[ 'class' ]} jp_simple_payments__wrapper">
+	<h2 class="jp_simple_payments__title">{$data['title']}</h2>
+	<div class="jp_simple_payments__description">{$data['description']}</div>
+	<div class="jp_simple_payments__price">{$data['price']}</div>
+	<div class="jp_simple_payments__button" id="{$data['dom_id']}"></div>
+</div>
+TEMPLATE;
+		return $output;
+	}
+
+	function format_price( $price, $currency ) {
+		// TODO: better logic here.
+		return $price. " " . $currency;
+	}
+
+	/**
+	 * Allows custom post types to be used by REST API.
+	 * @param $post_types
+	 * @see hook 'rest_api_allowed_post_types'
+	 * @return array
+	 */
+	function allow_rest_api_types( $post_types ) {
+		$post_types[] = self::$post_type_order;
```

Comment:
Wasn't the `[]=` shorthand introduced in php 5.3? Maybe I'm wrong; I've always thought it was, but I'm having a hard time finding evidence. (If we are following WP core coding standards, we can't use features newer than PHP 5.2.)